src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        ensureCapacity();
+        elementData[size] = value;
+        size++;
+    }
+
+    private void ensureCapacity() {
+
+        if (size == elementData.length) {
+            grow();
+        }
```

Comment:
The `ensureCapacity` method contains an empty line at the beginning of its implementation which should be removed according to the checklist.